packages/flutter_tools/lib/src/artifacts.dart

```diff
@@ -33,7 +33,7 @@ String _artifactToFileName(Artifact artifact) {
     case Artifact.dartVmEntryPointsTxt:
       return 'dart_vm_entry_points.txt';
     case Artifact.genSnapshot:
-      return 'gen_snapshot';
+      return platform.isWindows ? 'gen_snapshot.exe' : 'gen_snapshot';
     case Artifact.flutterTester:
       return 'flutter_tester';
```

Comment:
Why doesn't flutter_tester need it?

Reply:
> Why doesn't flutter_tester need it?

1) I don't see flutter_tester allowed on Windows(line 281 below).
2) Only gen_snapshot has file existence check(line 269 below).